base/server/src/main/java/org/dogtagpki/server/cli/SubsystemIdGeneratorUpdateCLI.java

```diff
@@ -0,0 +1,310 @@
+            String port = entry.getAttribute("securePort").getStringValues().nextElement();
+            BigInteger next = new BigInteger(endRange, 16);
+            if (lastUsedSerial.compareTo(next) < 0) {
+                lastUsedSerial = next;
+                nextRangeToUpdate = host.equals(hostName) && port.equals(securePort);
+                
+            }
+        }
+
+        if (nextRangeToUpdate) {
+            // nextRange is updated using last range entry or, if no ranges, the configured endSerialNumber
+            if (lastUsedSerial == BigInteger.ZERO) {
+                lastUsedSerial = new BigInteger(configEndSerialNumber, 16);
+            }
+            BigInteger nextSerialNumber = lastUsedSerial.add(BigInteger.ONE);
+            String serialDN = dbConfig.getSerialDN() + "," + baseDN;
+            // store nextRange as decimal
+            LDAPAttribute attrSerialNextRange = new LDAPAttribute("nextRange", nextSerialNumber.toString());
+
+            LDAPModification serialmod = new LDAPModification(LDAPModification.REPLACE, attrSerialNextRange);
+
+            conn.modify(serialDN, serialmod);
+        }
+
+        LDAPSearchResults instanceRanges = conn.search(rangeDN, LDAPv3.SCOPE_SUB, "(&(objectClass=pkiRange)(host= " +
+                    hostName + ")(SecurePort=" + securePort + "))", null, false);
+        
+        // update all ranges associated to the CA to update to decimal
+        while (instanceRanges.hasMoreElements()) {
+            LDAPEntry entry = instanceRanges.next();
+            String beginRange = entry.getAttribute("beginRange").getStringValues().nextElement();
+            BigInteger beginRangeNo = new BigInteger(beginRange, 16);
+            String endRange = entry.getAttribute("endRange").getStringValues().nextElement();
+            BigInteger endRangeNo = new BigInteger(endRange, 16);
+            LDAPAttributeSet attrs = new LDAPAttributeSet();
+            attrs.add(new LDAPAttribute("objectClass", "top"));
+            attrs.add(new LDAPAttribute("objectClass", "pkiRange"));
+
+            // store beginRange as decimal
+            attrs.add(new LDAPAttribute("beginRange", beginRangeNo.toString()));
+
+            // store endRange as decimal
+            attrs.add(new LDAPAttribute("endRange", endRangeNo.toString()));
+
+            attrs.add(new LDAPAttribute("cn", beginRangeNo.toString()));
+            attrs.add(new LDAPAttribute("host", hostName));
+            attrs.add(new LDAPAttribute("securePort", securePort));
+
+            String dn = "cn=" + beginRangeNo.toString() + "," + rangeDN;
+            LDAPEntry rangeEntry = new LDAPEntry(dn, attrs);
+            logger.info("SubsystemRangeGeneratorUpdateCLI.updateRanges: Remove entry " + entry.getDN());
+            conn.delete(entry.getDN());
+            logger.info("SubsystemRangeGeneratorUpdateCLI.updateRanges: Adding entry " + dn);
+            conn.add(rangeEntry);
```

Comment:
Here we're deleting the old `cn=<hex>` and adding a new `cn=<decimal>`. Is there a possibility that the `cn=<decimal>` will conflict with an existing range?

Reply:
I was trying to compute this to exclude this possibility but I cannot. However, since the conversion between hex and dec is not linear, the results are not multiple and the value increase very quickly I cannot find combination of value where overlap could happen so I consider very unlikely. However, we could try to find if a combination is possible orwe could add a check and eventually do a correction (e.g. adding or reducing the range by one entry)